### PR TITLE
Search tour

### PR DESCRIPTION
**Pull request by nguyenduyson11**
ticket
#34463 (https://edu-redmine.sun-asterisk.vn/issues/34463/edit)
What PR do?
Users can search for tours by name

Ticket pub backlog
none
envident
image

### CODE REVIEW

**Comment by nguyenduyson11** [comment]
ready

File: app/controllers/tours_controller.rb [hunk 1]
```diff
@@ -21,4 +18,11 @@ def find_tour
     flash[:error] = t "not_find_tour"
     redirect_to root_path
   end
+
+  def search
+    @tours =  paginate(Tour.sort_tours)
+    return if params[:key].blank?
+
+    @tours = paginate(Tour.search_tour(params[:key]))
```
Comment (vanvtt-0952) [inline app/controllers/tours_controller.rb]: + @tours = if params[:key].blank? paginate(Tour.sort_tours) : paginate(Tour.search_tour(params[:key]))

+ return  @tours = paginate(Tour.sort_tours) if params[:key].blank?
@tours = paginate(Tour.search_tour(params[:key]))

+
@tours = paginate(Tour.search_tour(params[:key])) 
check key ở model

+
@tours = if params[:key].blank? 
Tour.search_tour(params[:key])
else
Tour
end.sortxxx..paginate...

**Comment by nguyenduyson11** [comment]
ready

**Comment by nguyenduyson11** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented

**Comment by nguyenduyson11** [comment]
ready

File: app/controllers/tours_controller.rb [hunk 2]
```diff
@@ -21,4 +18,12 @@ def find_tour
     flash[:error] = t "not_find_tour"
     redirect_to root_path
   end
+
+  def search
+    @tours = paginate(if params[:key].blank?
```
Comment (tham-1781) [inline app/controllers/tours_controller.rb]:
```suggestion
    tours = if params[:key].blank?
    Tour
                      else
                        Tour.search_tour(params[:key])
                      end.newest
    @tours = paginate tours
```

File: app/models/tour.rb [hunk 3]
```diff
@@ -2,5 +2,8 @@ class Tour < ApplicationRecord
   has_many :book_tours, dependent: :destroy
   belongs_to :tour_type
   scope :tours_Highlight, ->{joins(:book_tours).group(:tour_id).last(4)}
-  scope :sort_tours, ->{order(created_at: :desc)}
+  scope :sort_tours_day_desc, ->{order(created_at: :desc)}
```
Comment (tham-1781) [inline app/models/tour.rb]:
```suggestion
  scope :newest, ->{order(created_at: :desc)}
```

**Comment by nguyenduyson11** [comment]
ready

**Comment by nguyenduyson11** [comment]
ready

**Comment by nguyenduyson11** [comment]
ready

**Pull request merged**